app/script/client/ClientRepository.coffee

```diff
@@ -534,45 +506,29 @@ class z.client.ClientRepository
     throw new z.client.ClientError z.client.ClientError::TYPE.NO_CLIENT_ID if not client_id
     return user_id is @self_user().id and client_id is @current_client().id
 
-  ###
-  Remove a client from the local clients.
-  @private
-  @param client_id [String] ID of client to be removed
-  ###
-  _remove_client: (client_id) =>
-    for client in @clients() when client.id is client_id
-      @clients.remove client
-      break
-
 
   ###############################################################################
   # Conversation Events
   ###############################################################################
 
   ###
   A client was added by the self user.
-  @todo map, save and add to user
   @param event_json [Object] JSON data of 'user.client-add' event
   ###
   on_client_add: (event_json) =>
     @logger.log @logger.levels.INFO, 'Client of self user added', event_json
-    amplify.publish z.event.WebApp.SELF.CLIENT_ADD, event_json.client
+    client_et = @client_mapper.map_client event_json.client
+    amplify.publish z.event.WebApp.CLIENT.ADD, @self_user().id, client_et
 
   ###
-  A client was added by the self user.
+  A client was removed by the self user.
   @param event_json [Object] JSON data of 'user.client-remove' event
   ###
   on_client_remove: (event_json) =>
     client_id = event_json?.client.id
-    @_client_removal client_id if client_id
+    return if not client_id
 
-  ###
-  Remove a client of the self user identified by id.
-  @private
-  @param client_id [String] ID of client to be removed
-  ###
-  _client_removal: (client_id) ->
-    @remove_client client_id
-    .then =>
-      @logger.log "Removed client from database: #{client_id}"
-      amplify.publish z.event.WebApp.SIGN_OUT, 'client_removed', true if client_id is @current_client().id
+    if client_id is @current_client().id
+      amplify.publish z.event.WebApp.SIGN_OUT, 'client_removed', true
```

Comment:
String constant for logout reasons would be nice. Because there is now `'user_requested'`, `'session_expired'` and `'client_removed'`.